src/sync/PostSyncProcessor.js

```diff
@@ -0,0 +1,157 @@
+/**
+ * All utility functions for processing synced records after being synced. These should changes
+ * should be detected by SyncQueue and sent back to the server on next sync.
+ */
+
+import { getNextNumber, NUMBER_SEQUENCE_KEYS } from '../database/utilities';
+
+const { REQUISITION_SERIAL_NUMBER, SUPPLIER_INVOICE_NUMBER } = NUMBER_SEQUENCE_KEYS;
+
+export class PostSyncProcessor {
+  constructor(database) {
+    this.database = database;
+    this.recordQueue = new Map(); // Map of [recordId, recordType]
```

Comment:
Yea insertion order is guaranteed (key advantage here) and the object methods are pretty semantic.

`myMap.has(thisKey)` vs `!myObject[thiskey]`
`myMap.remove(thisKey)` vs ~~I guess making a new object with all the same properties except the one to remove~~ [Wow I never knew!](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete) `delete myObject[thiskey]`